test/jdk/java/net/httpclient/HttpRequestNewBuilderTest.java

```diff
@@ -0,0 +1,272 @@
+   @Test(dataProvider = "testRequests")
+   public void testPOST(HttpRequest request) {
+       var r = HttpRequest.newBuilder(request)
+               .POST(HttpRequest.BodyPublishers.ofString("testData"))
+               .build();
+       assertEquals(r.method(), "POST");
+       assertTrue(r.bodyPublisher().isPresent());
+       assertEquals(r.bodyPublisher().get().contentLength(), 8);
+       assertAllOtherElementsEqual(r, request, "method");
+
+       testBodyPublisher("POST", request);
+   }
+
+   @Test(dataProvider = "testRequests")
+   public void testPUT(HttpRequest request) {
+       var r = HttpRequest.newBuilder(request)
+               .PUT(HttpRequest.BodyPublishers.ofString("testData"))
+               .build();
+       assertEquals(r.method(), "PUT");
+       assertTrue(r.bodyPublisher().isPresent());
+       assertEquals(r.bodyPublisher().get().contentLength(), 8);
+       assertAllOtherElementsEqual(r, request, "method");
+
+       testBodyPublisher("PUT", request);
+   }
+
+   @Test(dataProvider = "testRequests")
+   public void testUserDefinedMethod(HttpRequest request) {
+       testBodyPublisher("TEST", request);
+   }
+
+   @Test
+   public void testInvalidMethod() throws URISyntaxException {
+       URI testURI = new URI("http://www.foo.com/");
+       var r = new HttpRequest() {
+           @Override
+           public Optional<BodyPublisher> bodyPublisher() { return Optional.empty(); }
+           @Override
+           public String method() { return "CONNECT"; }
+           @Override
+           public Optional<Duration> timeout() { return Optional.empty(); }
+           @Override
+           public boolean expectContinue() { return false; }
+           @Override
+           public URI uri() { return testURI; }
+           @Override
+           public Optional<Version> version() { return Optional.empty(); }
+           @Override
+           public HttpHeaders headers() { return HttpHeaders.of(Map.of(), (x,y) -> true); }
+       };
+       assertThrows(IAE, () -> HttpRequest.newBuilder(r).build());
+   }
+
+   @Test
+   public void testInvalidURI() throws URISyntaxException {
+       // invalid URI scheme
+       URI badURI = new URI("ftp://foo.com/somefile");
+       var r = new HttpRequest() {
+           @Override
+           public Optional<BodyPublisher> bodyPublisher() { return Optional.empty(); }
+           @Override
+           public String method() { return "GET"; }
+           @Override
+           public Optional<Duration> timeout() { return Optional.empty(); }
+           @Override
+           public boolean expectContinue() { return false; }
+           @Override
+           public URI uri() { return badURI; }
+           @Override
+           public Optional<Version> version() { return Optional.empty(); }
+           @Override
+           public HttpHeaders headers() { return HttpHeaders.of(Map.of(), (x,y) -> true); }
+       };
+       assertThrows(IAE, () -> HttpRequest.newBuilder(r).build());
+   }
+}
```

Comment:
It is customary to have a new line at the end of files.

Reply:
New line added at end of file. See commit https://github.com/openjdk/jdk/pull/1059/commits/7928083ae954c44b5357064b86af0e1a3f53588b